src/components/TodoList/TodoList.js

```diff
@@ -0,0 +1,165 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
+/* eslint-disable jsx-a11y/label-has-for */
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import TodoItem from '../TodoItem/TodoItem';
+import './TodoList.css';
+
+class TodoList extends Component {
+  state = {
+    filter: 'all',
+  }
+
+  handleClickButtonClear = () => {
+    const { items, removeTodos } = this.props;
+
+    removeTodos(items.filter(item => item.completed).map(item => item.id));
+  }
+
+  handleFilters = ({ target }) => {
+    const filter = target.getAttribute('href').slice(2) || 'all';
+
+    this.setState({
+      filter,
+    });
+  }
+
+  handleToogleAll = ({ target }) => {
+    const { checked } = target;
+    const { setCompletedAll } = this.props;
+
+    setCompletedAll(checked);
+  }
+
+  render() {
+    const {
+      items,
+      changeCompleted,
+      removeTodos,
+      editTask,
+    } = this.props;
+    const { filter } = this.state;
+
+    const createFilteredTodos = (list, switcher) => {
```

Comment:
move this outside `render()`, because it is created after each rendering now

Reply:
> move this outside `render()`, because it is created after each rendering now

refactored